### PR TITLE
ICU-20978 Reduce unnecessary building for tools when cross-building

### PR DESCRIPTION
**Pull request by kateinoigakukun**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20978
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/configure.ac [hunk 1]
```diff
@@ -1143,6 +1143,8 @@ AC_ARG_ENABLE(fuzzer,
 	fuzzer=false)
 ICU_CONDITIONAL(FUZZER, test "$fuzzer" = true)
 
+ICU_CONDITIONAL(DATA, test "$tools" = true || test "$cross_compiling" = "yes")
```
Comment (srl295) [inline icu4c/source/configure.ac]: need to regenerate configure

Comment (kateinoigakukun) [reply icu4c/source/configure.ac]: Updated 👍

**Comment by kateinoigakukun** [comment]
@srl295 Could you take a look again? 🙏

**Comment by kateinoigakukun** [comment]
@srl295 Thanks! It's ready for merge but I have no permission to merge. Could you do that?

**Comment by kateinoigakukun** [comment]
Sorry for pinging you. Could you merge this?

**Pull request merged**